### PR TITLE
Fix various bounds

### PR DESCRIPTION
**Pull request by mseri**
Seen on #21399 and #21396

### CODE REVIEW

**Comment by mseri** [comment]
Failures are described in the commit messages

File: packages/bytearray/bytearray.1.0.1/opam [hunk 1]
```diff
@@ -11,7 +11,7 @@ depends: [
   "base-bigarray"
   "cppo"
```
Comment (mseri) [inline packages/bytearray/bytearray.1.0.1/opam:12]:
```suggestion
  "cppo" {>= "1.1.2"}
```
Uses `cppo -V`

File: packages/bytearray/bytearray.1.0.2/opam [hunk 2]
```diff
@@ -12,7 +12,7 @@ depends: [
   "base-bigarray"
   "cppo"
```
Comment (mseri) [inline packages/bytearray/bytearray.1.0.2/opam:13]:
```suggestion
  "cppo" {>= "1.1.2"}
```

File: packages/gufo/gufo.0.1.2/opam [hunk 3]
```diff
@@ -26,7 +26,7 @@ depends: [
   "lwt"
   "lambda-term" {>= "2.0.2"}
   "base-unix"
-  "sedlex" {>= "2.0.0"}
+  "sedlex" {>= "2.0.0" & < "3.0.0"}
```
Comment (mseri) [inline packages/gufo/gufo.0.1.2/opam:29]:
```suggestion
  "sedlex" {>= "2.0.0" & < "3.0"}
```

**Comment by jberdine** [comment]
Aside: Packages in opam-repository should not be allowed to specify `-Werror`. Is there some way to override such things globally in opam? Perhaps opam could set `OCAMLPARAM=_,warn-error=-a`.

**Comment by mseri** [comment]
I agree, but there is no global mechanism afaik. You can change them only creating manual patches and releasing a patched version

**Pull request merged**